prac_08/dynamic_labels.py

```diff
@@ -0,0 +1,27 @@
+from kivy.app import App
+from kivy.lang import Builder
+from kivy.uix.label import Label
+
+
+class DynamicLabelsApp(App):
+    """App for displaying dynamic labels for a list of names."""
+
+    def __init__(self, **name):
+        super().__init__(**name)
+        self.names = ['dog', 'Cat ', 'bird']
+
+    def build(self):
+        """Build the Kivy GUI."""
+        self.title = "Dynamic Labels"
+        self.root = Builder.load_file('dynamic_labels.kv')
+        self.create_names()
+        return self.root
+
+    def create_names(self):
+        """Create a label for each name."""
+        for name in self.names:
+            self.root.ids.names_box.add_widget(Label(text=name))
```

Comment:
This is the right code to make the labels/names.
You just need to copy this into the Build method.

Also this is what I did:
temp_label = Label(text=name)
self.root.ids.names_box.add_widget(temp_label)
            
I'm not sure which way is better, as the way you did your code is more concise, I'll have to ask Lindsay about this